File: 이것이 취업을 위한 코딩 테스트다/그리디/문자열_뒤집기/문자열_뒤집기.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
  string str;
  cin >> str;

  int count0 = 0;
  int count1 = 0;

  if (str[0] = '0') {
    count1 += 1;
  } else {
    count0 += 1;
  }

  for (int i = 1; i < str.size() - 1; i++) {
    if (str[i] != str[i+1]) {
      if (str[i+1] == '0') {
        count1 += 1;
      } else {
        count0 += 1;
      }
    }
  }

  cout << min(count0, count1);

  return 0;
}

File: 이것이 취업을 위한 코딩 테스트다/그리디/문자열_뒤집기/문자열_뒤집기.py
# s = input()
# current = s[0]
# obj = {
#   '0': 0,
#   '1': 0
# }

# if current == '0':
#   obj['1'] += 1
# else:
#   obj['0'] += 1

# for c in s:
#   if c == current:
#     continue;
#   else:
#     obj[c] += 1

#   current = c

# print(min(obj['0'], obj['1']))

data = input()
count_0 = 0
count_1 = 0
current = data[0]

if current == '0':
  count_1 += 1
else:
  count_0 += 1

for i in range(1, len(data)):
  if current != data[i]:
    if data[i] == '0':
      count_1 += 1
    else:
      count_0 += 1
    current = data[i]

print(min(count_0, count_1))